Code/GraphMol/DistGeomHelpers/Embedder.cpp

```diff
@@ -293,48 +438,49 @@ bool _minimizeWithExpTorsions(
   }
 
   // create the force field
-  ForceFields::ForceField *field;
-  if (useBasicKnowledge) {  // ETKDG or KDG
-    field = DistGeom::construct3DForceField(*mmat, positions3D, bonds, angles,
-                                            expTorsionAtoms, expTorsionAngles,
-                                            improperAtoms, atomNums);
+  std::unique_ptr<ForceFields::ForceField> field;
+  if (embedParams.useBasicKnowledge) {  // ETKDG or KDG
+    field.reset(DistGeom::construct3DForceField(
+        *eargs.mmat, positions3D, eargs.etkdgDetails->bonds,
+        eargs.etkdgDetails->angles, eargs.etkdgDetails->expTorsionAtoms,
+        eargs.etkdgDetails->expTorsionAngles, eargs.etkdgDetails->improperAtoms,
+        eargs.etkdgDetails->atomNums));
   } else {  // plain ETDG
-    field = DistGeom::constructPlain3DForceField(*mmat, positions3D, bonds,
-                                                 angles, expTorsionAtoms,
-                                                 expTorsionAngles, atomNums);
+    field.reset(DistGeom::constructPlain3DForceField(
+        *eargs.mmat, positions3D, eargs.etkdgDetails->bonds,
+        eargs.etkdgDetails->angles, eargs.etkdgDetails->expTorsionAtoms,
+        eargs.etkdgDetails->expTorsionAngles, eargs.etkdgDetails->atomNums));
```

Comment:
I think the construct3DForceField is only used for ETKDG, so maybe one could just pass on the CrystalFFdetails object

Reply:
good point. I just pushed those changes.